### PR TITLE
gha: Enable install kbs and coco components for TDX, but still skip the CDH test

### PR DESCRIPTION
**Pull request by GabyCT**
This PR enables the installation and unistallation of the kbs client as well as general coco components needed for the TDX GHA CI.

### CODE REVIEW

File: tests/integration/kubernetes/confidential_kbs.sh [hunk 1]
```diff
@@ -319,7 +319,10 @@ kbs_k8s_svc_host() {
 			-o jsonpath='{.spec.rules[0].host}' 2>/dev/null
 	elif kubectl get svc kbs-nodeport -n "$KBS_NS" &>/dev/null; then
 			local host
-			host=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[?(@.type=="ExternalIP")].address}' -n "$KBS_NS")
+			host=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[0].address}')
+			# I am not sure why we'd use the External / Internal IP as shown below,
```
Comment (wainersm) [inline tests/integration/kubernetes/confidential_kbs.sh:323]: IIRC this was need to proper work with kcli. @ldoktor should know why for sure

Comment (ldoktor) [reply tests/integration/kubernetes/confidential_kbs.sh]: On some versions (kcli cluster) the externalIP/address was not set while InternalIP was. So yes, this is necessary. Thanks for keeping it.

File: tests/integration/kubernetes/confidential_kbs.sh [hunk 2]
```diff
@@ -321,6 +321,8 @@ kbs_k8s_svc_host() {
 			local host
 			host=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[?(@.type=="ExternalIP")].address}' -n "$KBS_NS")
 			[ -z "$host"] && host=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}' -n "$KBS_NS")
+			hostname_i=$(hostname -i)
+			[ "$host" = "$hostname_i" ] && host="localhost"
```
Comment (ldoktor) [inline tests/integration/kubernetes/confidential_kbs.sh:326]: `[[ $(hostname -I) =~ "$host" ]]` would be IMO safer

File: tests/integration/kubernetes/confidential_kbs.sh [hunk 3]
```diff
@@ -139,7 +139,7 @@ kbs_install_cli() {
 	# Mininum required version to build the client (read from versions.yaml)
 	local rust_version
 	ensure_yq
-	rust_version=$(get_from_kata_deps "externals.coco-kbs.toolchain")
+	rust_version=$(get_from_kata_deps "externals.coco-trustee.toolchain")
```
Comment (ldoktor) [inline tests/integration/kubernetes/confidential_kbs.sh:142]: The change is fine, but the commit message is really cryptic. How about `externals.coco-kbs.toolchain` is not defined, get the rust_version from `externals.coco-trustee.toolchain`?

File: tests/integration/kubernetes/confidential_kbs.sh (hunk 2 above)
Comment (fidencio) [reply tests/integration/kubernetes/confidential_kbs.sh:326]: Will be fixed in the next iteration.

File: tests/integration/kubernetes/confidential_kbs.sh (hunk 3 above)
Comment (fidencio) [reply tests/integration/kubernetes/confidential_kbs.sh:142]: Ditto!

File: tests/integration/kubernetes/confidential_kbs.sh [hunk 4]
```diff
@@ -321,6 +321,7 @@ kbs_k8s_svc_host() {
 			local host
 			host=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[?(@.type=="ExternalIP")].address}' -n "$KBS_NS")
 			[ -z "$host"] && host=$(kubectl get nodes -o jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}' -n "$KBS_NS")
+			[[ "$(hostname -i)" =~ "$host" ]] && host="localhost"
```
Comment (ldoktor) [inline tests/integration/kubernetes/confidential_kbs.sh:335]: It should be capital `I` to check against all host IP addresses, not just the first one

Comment (fidencio) [reply tests/integration/kubernetes/confidential_kbs.sh]: I ended up dropping this commit.

File: versions.yaml [hunk 5]
```diff
@@ -227,9 +227,9 @@ externals:
   coco-trustee:
     description: "Provides attestation and secret delivery components"
     url: "https://github.com/confidential-containers/trustee"
-    version: "18c8ee378c6d83446ee635a702d5dee389028d8f"
+    version: "6adb8383309cbb7279f1d8e1e4620556ac66481e"
```
Comment (fitzthum) [inline versions.yaml:230]: There are newer hashes available. Any reason to choose this one in particular?

**Review by fitzthum — APPROVED** [review]
LGTM. Nice improvements.

**Review by wainersm — APPROVED** [review]
Thanks @GabyCT !

File: versions.yaml (hunk 5 above)
Comment (fidencio) [reply versions.yaml:230]: @fitzthum, I've decided to follow the rule of "bumping to use what we need", as we needed the kustomization for TDX.

**Review by dborquez — APPROVED** [review]
LGTM, thanks @GabyCT

**Comment by fidencio** [comment]
WIP check is stuck, I'm bypassing it.

**Pull request merged**